src/gui/doAction.cpp

```diff
@@ -883,6 +883,37 @@ void FurnaceGUI::doAction(int what) {
     case GUI_ACTION_WAVE_LIST_SAVE_RAW:
       if (curWave>=0 && curWave<(int)e->song.wave.size()) openFileDialog(GUI_FILE_WAVE_SAVE_RAW);
       break;
+    case GUI_ACTION_WAVE_LIST_CREATE_SAMPLE:
+      if (curWave>=0 && curWave<(int)e->song.wave.size()) {
+        curSample=e->addSample();
+        if (curSample==-1) {
+          showError(_("too many samples!"));
+        } else {
+          e->lockEngine([this]() {
+            DivSample* sample=e->getSample(curSample);
+            if (sample!=NULL) {
+              unsigned int waveLen=e->song.wave[curWave]->len;
+              sample->rate=(int)round(261.343f*waveLen); // c3
+              sample->centerRate=(int)round(261.343f*waveLen); // c3
```

Comment:
Remove "f". We can use higher precision.
The result will be truncated to int anyway.